Integrate with CircleCI

ynamiki opened the pull request:
Integrate with CircleCI:

https://circleci.com/gh/nec-blockchain/minbft

Commit:
Integrate with CircleCI

This resolves #7.

Signed-off-by: Yuta Namiki <y-namiki@ak.jp.nec.com>

sergefdrv reviewed (commented):
Would you mind adding a build status badge to the Readme?
```
[![CircleCI](https://circleci.com/gh/nec-blockchain/minbft/tree/master.svg?style=svg)](https://circleci.com/gh/nec-blockchain/minbft/tree/master)
```

Commit:
Add a CircleCI status badge to README

Signed-off-by: Yuta Namiki <y-namiki@ak.jp.nec.com>

ynamiki commented:
I have added a build status badge to the README.

CircleCI supports [custom Docker images](https://circleci.com/docs/2.0/custom-images/) so it is better to build a custom image with Intel SGX SDK and Go on Ubuntu 16.04 than to do everything in `.circleci/config.yml`. A problem on using a custom image is that we need to have a Docker Registry or to manage an account of Docker Hub. This enhancement is future work.

sergefdrv commented on .circleci/config.yml:
```diff
@@ -0,0 +1,44 @@
+            echo '. $SGX_SDK/environment' >> $BASH_ENV
+            echo 'export PATH=$PATH:$SGX_SDK/bin:$SGX_SDK/bin/x64' >> $BASH_ENV
+            echo 'export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$SGX_SDK/pkgconfig' >> $BASH_ENV
+            echo 'export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$SGX_SDK/sdk_libs' >> $BASH_ENV
```
`. $SGX_SDK/environment` should update `PATH`, `PKG_CONFIG_PATH`, and `LD_LIBRARY_PATH` environment variables accordingly. Do we really need to duplicate that in lines 28-30?

sergefdrv commented:
@ynamiki Thanks a lot for taking care of this! This CircleCI integration looks good and should suffice for now. I see how using custom Docker images will make CI more efficient. Let's give it a try later.

Commit:
Delete Duplicate Definition on Environment Variables

PATH, PKG_CONFIG_PATH, and LD_LIBRARY_PATH are defined in
$SGX_SDK/environmet.

Signed-off-by: Yuta Namiki <y-namiki@ak.jp.nec.com>

sergefdrv reviewed (approved):
The last commit could be squashed with the first one, but never mind, it's not a big deal :slightly_smiling_face:

sergefdrv commented:
@ynamiki I see "You’re not authorized to merge this pull request." That means I couldn't merge this pull request according to #12

ynamiki commented:
@sergefdrv Sorry, I'll check settings.

ynamiki commented:
@sergefdrv I have fixed the permission. Could you try again?
